### PR TITLE
Crystallo <-> Roof Crystallo on levelflip

### PR DESCRIPTION
**Pull request by Narre**
This PR makes Crystallos transform into Roof Crystallos on levelflip() (and vice versa).
Awaken Sleeping Crystallos get transformed into Roof Crystallos too.
Asleep Sleeping Crystallos get their texture flipped.
If awaken during a levelflip(), the Sleeping Crystallo will transform into a Roof Crystallo and immediately drop/attack.

This PR also combines `crystallo.sprite`, `scrystallo.sprite` and `rcrystallo.sprite` into one (`crystallo.sprite`) sprite file. This is done so that the crystallos can keep their sprite during the transformations, i.e. compatibility with custom sprites.

https://github.com/SuperTux/supertux/assets/14074789/5cb009a0-15e6-4b6b-82c4-68b67378c7ea

### CODE REVIEW

File: src/supertux/physic.hpp [hunk 1]
```diff
@@ -68,6 +68,8 @@ class Physic final
   /** Set gravity modifier factor to apply to object when enabled */
   void set_gravity_modifier(float modifier) { gravity_modifier = modifier; }
 
+  float get_gravity_modifier() { return gravity_modifier; }
```
Comment (tobbi) [inline src/supertux/physic.hpp:71]: vont

**Review by mrkubax10 — APPROVED** [review]

File: src/badguy/crystallo.hpp [hunk 2]
```diff
@@ -24,6 +24,7 @@ class Crystallo final : public WalkingBadguy
 {
 public:
   Crystallo(const ReaderMapping& reader);
+  Crystallo(Vector pos, Vector start_pos, float vel_x, std::unique_ptr<Sprite> sprite, Direction dir, float radius, std::string& script);
```
Comment (Vankata453) [inline src/badguy/crystallo.hpp:27]: `const Vector&`, `const std::string&`. Parameter list can be split into 2 lines (same for the definition in the `.cpp` file).

File: src/badguy/rcrystallo.hpp [hunk 3]
```diff
@@ -23,6 +23,7 @@ class RCrystallo final : public WalkingBadguy
 {
 public:
   RCrystallo(const ReaderMapping& reader);
+  RCrystallo(Vector pos, Vector start_pos, float vel_x, std::unique_ptr<Sprite> sprite, Direction dir, float radius, std::string& script, bool fall = false);
```
Comment (Vankata453) [inline src/badguy/rcrystallo.hpp:26]: `const Vector&`, `const std::string&`. Parameter list can be split in multiple lines (same for the definition in the `.cpp` file).

File: src/badguy/scrystallo.hpp [hunk 4]
```diff
@@ -48,12 +51,15 @@ class SCrystallo final : public WalkingBadguy
...
-  SCrystalloState state;
+  SCrystalloState m_state;
 
 private:
+  bool m_roof;
   float m_radius;
   float m_range;
   Vector m_radius_anchor;
+
+
```
Comment (Vankata453) [inline src/badguy/scrystallo.hpp:62]: Unneeded line.

**Review by tobbi — APPROVED** [review]

**Pull request merged**